### PR TITLE
(refactor): Use Nitro web server

### PR DESCRIPTION
**Pull request by zisra**
Closes [#424](https://github.com/movie-web/movie-web/issues/424)

- [x] Use [Nitro](https://nitro.unjs.io/) so you can deploy on any platform
- [x] Fix redirects not working
- [x] Fix body not working
- [x] Docker setup
- [x] Eslint action

### CODE REVIEW

**Comment by zisra** [comment]
Welp, can't find a way to fix [actions](https://github.com/zisra/simple-proxy/actions/runs/6128985110/job/16636637820#step:6:5)

File: .github/workflows/cloudflare.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  repository_dispatch:
+
+jobs:
+  ci:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
+        node: [ 14 ]
```
Comment (binaryoverload) [inline .github/workflows/cloudflare.yml:15]: The reason the action might be failing is due to the node version, this is out of date. 18 should be used instead.

File: routes/index.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    }`,
+  );
+
+  // Create a new mutable request object for the destination
+  const request = new Request(requestData.destination, {
+    headers: requestData.headers,
+    method: requestData.method,
+    body: requestData.body,
+  });
+  request.headers.set("Origin", new URL(requestData.destination).origin);
```
Comment (binaryoverload) [inline routes/index.ts:37]:
```suggestion

  request.headers.set("Host", new URL(requestData.destination).host);
  request.headers.set("Origin", new URL(requestData.destination).origin);
```

**Review by binaryoverload — CHANGES_REQUESTED** [review]
When refactoring a codebase, please make sure that you keep the formatting and dev dependencies consistent. 

Please delete all the GitHub actions, we will do them ourselves as it's a bit complicated to setup :)

**Review by binaryoverload — CHANGES_REQUESTED** [review]
This still does not work on Node.JS - please look at using the [proxyRequest](https://www.jsdocs.io/package/h3#proxyRequest) helper method as discussed.

File: package.json [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
+    "dev": "nitropack dev",
+    "build": "nitropack build",
+    "preview": "node .output/server/index.mjs",
     "lint": "eslint --ext .js src/",
-    "lint:fix": "eslint --fix --ext .js src/",
-    "lint:report": "eslint --ext .js --output-file eslint_report.json --format json src/"
+    "lint:fix": "eslint --fix --ext .js src/"
+  },
+  "dependencies": {
+    "eslint-plugin-prettier": "^5.0.0",
```
Comment (binaryoverload) [inline package.json:14]: This needs to be a dev dependency

**Review by binaryoverload — CHANGES_REQUESTED** [review]

**Comment by mrjvs** [comment]
@binaryoverload can you rereview this?
I took over from this PR, so most code is mine.
The todos in the readme are not in scope for this PR (and im lazy, dont wanna do it)

**Pull request merged**